src/print.c

```diff
@@ -245,6 +255,190 @@ launch_preview (const char *filename,
   return retval;
 }
 
+static int
+pdf_get_actions_page (char        *filename,
+                      PDF_ACTIONS  act,
+                      int          page)
+{
+  GSubprocess *process;
+  GInputStream *stream;
+  char *option1 = NULL;
+  char *option2 = NULL;
+  char buffer[50] = { 0 };
+
+  option1 = g_strconcat ("--file=", filename, NULL);
+  switch(act)
+  {
+    case PDF_TEST :
+       (void)page;
+       option2 = g_strdup ("--test");
+       break;
+    case PDF_NUM_PAGES :
+       (void)page;
+       option2 = g_strdup ("--pages");
+       break;
+    case PDF_WIDTH :
+       option2 = g_strdup_printf ("--width=%d", page);
+       break;
+    case PDF_HEIGHT :
+       option2 = g_strdup_printf ("--height=%d", page);
+       break;
+   default:
+       return 0;
+  }
+  process = g_subprocess_new(G_SUBPROCESS_FLAGS_STDOUT_PIPE, NULL, "/usr/libexec/pdftoraw", option1, option2, NULL);
```

Comment:
- This still needs to be sandboxed with `bwrap`.
- Don't hardcode `/usr/libexec`, this should be defined at compile time (`-DLIBEXEC_DIR=...`).

Reply:
> Don't hardcode /usr/libexec, this should be defined at compile time (-DLIBEXEC_DIR=...).

Fixed by 7ec9062903cce6